File: fathersDay/math/Day4 (633. Sum of Square Numbers)/Solution.py
class Solution:
    def sumOfSquareNumbers(self, c: int) -> bool:
        left, right = 0, int(c**0.5)
        while left <= right:
            current_sum = left * left + right * right
            if current_sum == c:
                return True
            elif current_sum < c:
                left += 1
            else:
                right -= 1
        return False
    
# Example usage
solution = Solution()
print(solution.sumOfSquareNumbers(5))  

File: fathersDay/math/Day4 (633. Sum of Square Numbers)/Solution.java
class Solution {
    public boolean judgeSquareSum(int c) {
        long left = 0;
        long right = (long) Math.sqrt(c);

        while (left <= right) {
            long currentSum = left * left + right * right;
            if (currentSum == c) {
                return true;
            } else if (currentSum < c) {
                left++;
            } else {
                right--;
            }
        }
        return false;
    }
}
